hazelcast/src/main/java/com/hazelcast/sql/impl/expression/string/SubstringFunction.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.TriExpression;
+import com.hazelcast.sql.impl.expression.math.MathFunctionUtils;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+public class SubstringFunction extends TriExpression<String> implements IdentifiedDataSerializable {
+    public SubstringFunction() {
+        // No-op
+    }
+
+    private SubstringFunction(Expression<?> input, Expression<?> start, Expression<?> length) {
+        super(input, start, length);
+    }
+
+    public static SubstringFunction create(Expression<?> input, Expression<?> start, Expression<?> length) {
+        return new SubstringFunction(input, start, length);
+    }
+
+    @SuppressWarnings({"checkstyle:CyclomaticComplexity", "checkstyle:NPathComplexity"})
+    @Override
+    public String eval(Row row, ExpressionEvalContext context) {
+        // Get input
+        String input;
+
+        try {
+            input = StringExpressionUtils.asVarchar(operand1, row, context);
+        } catch (Exception e) {
+            // Conversion to String failed. E.g. NPE on UserClass.toString()
+            throw QueryException.dataException("Failed to get value of input operand of SUBSTRING function", e);
+        }
+
+        if (input == null) {
+            // NULL always yields NULL
+            return null;
+        }
+
+        Integer start;
+
+        if (operand2 != null) {
+            start = MathFunctionUtils.asInt(operand2, row, context);
+
+            if (start == null) {
+                return null;
+            }
+        } else {
+            start = null;
```

Comment:
This line is not covered by integration test. Actually I can't figure out how can you pass null `operand2`.